src/Logging/Logging.Abstractions/src/IScopeFuncLogger.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    public interface IScopeFuncLogger
+    {
+        IDisposable BeginScope<TState, TScopeState>(Func<TState, TScopeState> scopeCreator, in TState state);
```

Comment:
Could be a default interface method on ILogger; however as its `netstandard2.0` I guess that wouldn't work?

#ifdef an interface? 🤔